src/main.ts

```diff
@@ -124,9 +124,10 @@ async function run(): Promise<void> {
     }
 
     try {
-        const checkRunPayload = utils.getCheckRunPayloadWithScanResult(trivyStatus, dockleStatus);
+        const checkRunPayload = utils.getCheckRunPayload(trivyStatus, dockleStatus);
+        const checkRunThroughAppPayload = utils.getCheckRunThroughAppPayload(trivyStatus, dockleStatus);
         const githubClient = new GitHubClient(process.env.GITHUB_REPOSITORY, inputHelper.githubToken);
-        await githubClient.createCheckRun(checkRunPayload);
+        await githubClient.createCheckRunWithFallback(checkRunThroughAppPayload, checkRunPayload);
```

Comment:
It's not the client's duty to worry about fall back ideally. The validation of status logic should be here.

Reply:
Moved the logic to utils.